exporters/stats/prometheus/src/main/java/io/opencensus/exporter/stats/prometheus/PrometheusStatsCollector.java

```diff
@@ -63,27 +72,49 @@ public static void createAndRegister() {
           samples.add(PrometheusExportUtils.createMetricFamilySamples(viewData));
         }
       }
+      span.addAnnotation("Finish collecting Prometheus Metric Samples.");
     } catch (Throwable e) {
       logger.log(Level.WARNING, "Exception thrown when collecting metric samples.", e);
+      span.setStatus(
+          Status.UNKNOWN.withDescription(
+              "Exception thrown when collecting Prometheus Metric Samples: "
+                  + exceptionMessage(e)));
     }
+    span.end();
```

Comment:
This should be moved in a finally block.

Reply:
Done.